c_emulator/riscv_sim.c

```diff
@@ -281,9 +286,15 @@ static int process_args(int argc, char **argv)
       fprintf(stderr, "enabling misaligned access.\n");
       rv_enable_misaligned = true;
       break;
-    case 'P':
-      fprintf(stderr, "enabling PMP support.\n");
-      rv_enable_pmp = true;
+    case OPT_PMP_COUNT:
+      pmp_count = atol(optarg);
+      fprintf(stderr, "PMP count: %lld\n", pmp_count);
+      rv_pmp_count = pmp_count;
```

Comment:
Do we need to handle invalid `optarg` here?

Reply:
Ah yes the Sail assumes it is <=64 so we need to check that here. I'll add that.

Reply:
Done. I added a check for `pmp_grain` too.